### ISSUE TITLE
CA1508 dead code: false positive when checking an expression a second time

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#55622]
[This code](
https://github.com/aarnott/pclcommandbase/blob/91f346e07eb627a75f7f4f188afd4eb11721261b/test/PCLCommandBase.Tests/CommandBaseTests.cs#L154-L154) produces CA1508 (dead code) when not suppressed.

**Version Used**: VS 16.11 Preview 4 (31605.320.d16.11)

**Steps to Reproduce**:

```
git clone https://github.com/aarnott/pclcommandbase
cd pclcommandbase
git checkout 91f346e07eb627a75f7f4f188afd4eb11721261b
pclcommandbase.sln
```

**Expected Behavior**:

No warnings

**Actual Behavior**:

> Warning	CA1508	'(await eventsQueue.DequeueAsync(this.TimeoutToken)).PropertyName != nameof(this.command.IsExecuting)' is always 'false'. Remove or refactor the condition(s) to avoid dead code.	PCLCommandBase.Tests (net472)	D:\git\pclcommandbase\test\PCLCommandBase.Tests\CommandBaseTests.cs

I haven't been able to reproduce it in a simpler case.

**Comment by Youssef1313** [comment]
@jinujoseph @mavasani Can you move to roslyn-analyzers repo please?

@AArnott This specific analyzer currently have lots of false positive. Might be better to complete disable it for now. fwiw, @sharwell had recommended to keep it disabled even if those false positives are fixed.

From Roslyn Discord channel:

![image](https://user-images.githubusercontent.com/31348972/129457409-0733e31e-1c8f-4ffa-8f61-272ea1d3f9cf.png)
